lib/Sema/CSApply.cpp

```diff
@@ -5762,19 +5763,33 @@ Expr *ExprRewriter::coerceCallArguments(
     }
   }
 
+  // If we only have one argument and one parameter, we're done.
+  if (argTuple == nullptr &&
+      params.size() == 1 &&
+      params[0].getLabel().empty() &&
+      !params[0].isVariadic()) {
```

Comment:
The logic to check if parameter array is really just a single unlabeled parameter could be abstracted into a method on AnyFunctionType or something, because it's repeated in multiple places...

Reply:
You're right? Mind if I do it in a follow-up PR?

Reply:
Absolutely!